### PR TITLE
Feature/red 235 add partialcreditbestscore

### PR DESCRIPTION
**Pull request by PanopticaRising**

### CODE REVIEW

File: src/Courses/CourseDetailsTabs/GradeTable.tsx [hunk 1]
```diff
@@ -16,15 +17,11 @@ export const GradeTable: React.FC<GradeTableProps> = ({courseName, grades, onRow
...
+        ...(_.isNil(obj.average) ? undefined : {average: obj.average.toPercentString()}),
+        ...(_.isNil(obj.effectiveScore) ? undefined : {effectiveScore: obj.effectiveScore.toPercentString()}),
+        ...(_.isNil(obj.systemScore) ? undefined : {systemScore: obj.systemScore.toPercentString()}),
     }));
     const headers = _(safeGrades[0]).keys().filter(n => n !== 'id').value();
-    if(headers.indexOf('average') >= 0) {
-        // Would include this in above mapping, however using ternary operated resulted in an empty column in questions
-        safeGrades = safeGrades.map(obj => ({
-            ...obj,
-            average: _.isNil(obj.average) ? '--' : `${(obj.average * 100).toFixed(1)}%`
```
Comment (red-tt-bear) [inline src/Courses/CourseDetailsTabs/GradeTable.tsx:25]: What happened to the `--`

Comment (red-tt-bear) [reply src/Courses/CourseDetailsTabs/GradeTable.tsx:25]: This happens if the weight is 0 therefore it would be divide by zero and it returns null

**Review by red-tt-bear — CHANGES_REQUESTED** [review]
What happens when the result of the query is null (more details on the -- comment

File: src/Courses/CourseDetailsTabs/GradeTable.tsx (hunk 1 above)
Comment (PanopticaRising) [reply src/Courses/CourseDetailsTabs/GradeTable.tsx:25]: "I mean optional" - @red-tt-bear 

I need to check both if the fields are present in the headers AND if they're null.

**Review by red-tt-bear — APPROVED** [review]

**Review by doombot-exe — APPROVED** [review]

**Pull request merged**